src/install/warmViews.js

```diff
@@ -0,0 +1,176 @@
+  if (!indexers || !indexers.length) {
+    return;
+  }
+
+  const logProgress = (indexer) => {
+    // progress bar stretches to match console width.
+    // 60 is roughly the nbr of chars displayed around the bar (ddoc name + debug padding)
+    const barLength = process.stdout.columns - 60,
+          progress = `${indexer.progress}%`,
+          filledBarLength = (indexer.progress / 100 * barLength),
+          bar = progress
+            .padStart((filledBarLength + progress.length) / 2, '|')
+            .padEnd(filledBarLength, '|')
+            .padEnd(barLength, '_'),
+          ddocName = indexer.design_document.padEnd(35, ' ');
+
+    debug(`${ddocName}[${bar}]`);
+  };
+
+  debug('View indexer progress');
+  indexers.forEach(logProgress);
+};
+
+// Groups tasks by `design_document` and calculates the average progress per ddoc
+// When a task is finished, it disappears from _active_tasks
+const updateIndexers = (deployDoc, runningTasks) => {
+  const entry = deployDoc.log[deployDoc.log.length - 1],
+        indexers = entry.indexers || [];
+
+  // We assume all previous tasks have finished.
+  indexers.forEach(setTasksToComplete);
+  // If a task is new or still running, it's progress is updated
+  updateRunningTasks(indexers, runningTasks);
+  indexers.forEach(calculateAverageProgress);
+
+  entry.indexers = indexers;
+
+  logIndexersProgress(indexers);
+  return utils.update(deployDoc);
+};
+
+const setTasksToComplete = (indexer) => {
+  Object
+    .keys(indexer.tasks)
+    .forEach(pid => {
+      indexer.tasks[pid] = 100;
+    });
+};
+
+const calculateAverageProgress = (indexer) => {
+  const tasks = Object.keys(indexer.tasks);
+  indexer.progress = Math.round(tasks.reduce((progress, pid) => progress + indexer.tasks[pid], 0) / tasks.length);
+};
+
+const updateRunningTasks = (indexers, activeTasks = []) => {
+  activeTasks.forEach(task => {
+    let indexer = indexers.find(indexer => indexer.design_document === task.design_document);
+    if (!indexer) {
+      indexer = {
+        design_document: task.design_document,
+        tasks: {},
+      };
+      indexers.push(indexer);
+    }
+    indexer.tasks[`${task.node}-${task.pid}`] = task.progress;
+  });
+};
+
+// FIXME: I'm pretty sure this doesn't work with Mango indexes
+// https://github.com/medic/horticulturalist/issues/39
+const firstView = ddoc =>
+  `${ddoc._id.replace('_design/', '')}/${Object.keys(ddoc.views).find(k => k !== 'lib')}`;
+
+const viewQueries = ddocs => ddocs
+      .filter(ddoc => ddoc.views && Object.keys(ddoc.views).length)
+      .map(firstView);
+
+module.exports = () => {
+  let stopViewWarming = false;
+
+  const progressLoop = (deployDoc, _queryInterval) => {
+    return new Promise((res, rej) => {
+      const checkProgressLater = (waitMs) => {
+        if (stopViewWarming) {
+          return res();
+        }
+
+        setTimeout(() => {
+          writeProgress(deployDoc)
+            .then(checkProgressLater)
```

Comment:
You should change this to:
```
.then(() => checkProgressLater())
```
otherwise checkProgressLater will receive the result from `writeProgress` as `waitMs` and query the db constantly.